CommandLine/testing/SimpleTests/file_input_output.sog/create.edl

```diff
@@ -0,0 +1,196 @@
+file_text_close(text_write);
+
+// TEXT APPEND
+var text_append;
+text_append = file_text_open_append(text_path);
+gtest_assert_ge(text_append, 0);
+file_text_write_real(text_append, 2);
+file_text_write_real(text_append, 5);
+file_text_write_string(text_append, " b");
+file_text_writeln(text_append);
+file_text_close(text_append);
+
+// TEXT READ
+var text_read;
+text_read = file_text_open_read(text_path);
+gtest_assert_ge(text_read, 0);
+gtest_expect_false(file_text_eof(text_read));
+gtest_expect_false(file_text_eoln(text_read));
+gtest_expect_eq(file_text_read_string(text_read),"apple pear bear");
+gtest_expect_false(file_text_eof(text_read));
+gtest_expect_true(file_text_eoln(text_read));
+file_text_readln(text_read);
+gtest_expect_eq(file_text_read_real(text_read),0);
+gtest_expect_eq(file_text_read_real(text_read),-1);
+gtest_expect_eq(file_text_read_real(text_read),253);
+gtest_expect_false(file_text_eof(text_read));
+gtest_expect_true(file_text_eoln(text_read));
+file_text_readln(text_read);
+file_text_read_real(text_read) // TODO test epsilon
+file_text_read_real(text_read) // TODO test epsilon
+file_text_read_real(text_read) // TODO test epsilon
+file_text_readln(text_read);
+gtest_expect_eq(file_text_read_real(text_read),2);
+gtest_expect_eq(file_text_read_string(text_read)," 5 b");
+file_text_readln(text_read);
+// Unix convention/end of file line is empty
+gtest_expect_false(file_text_eof(text_read));
+gtest_expect_true(file_text_eoln(text_read));
+file_text_readln(text_read);
+gtest_expect_true(file_text_eof(text_read));
+gtest_expect_true(file_text_eoln(text_read));
+file_text_close(text_read);
+
+/// General File Functions
+gtest_expect_false(file_exists("ENIGMA John Doe.txt"));
+gtest_expect_false(directory_exists("ENIGMA Folders"));
+file_text_close(file_text_open_write("ENIGMA John Doe.txt"));
+directory_create("ENIGMA Folders");
+gtest_expect_true(file_exists("ENIGMA John Doe.txt"));
+gtest_expect_true(directory_exists("ENIGMA Folders"));
+
+gtest_expect_false(file_exists("Games Are Fun.txt"));
+file_rename("ENIGMA John Doe.txt","Games Are Fun.txt");
+gtest_expect_false(file_exists("ENIGMA John Doe.txt"));
+gtest_expect_true(file_exists("Games Are Fun.txt"));
+
+gtest_expect_false(file_exists("Development Community.txt"));
+file_copy("Games Are Fun.txt","Development Community.txt");
+gtest_expect_true(file_exists("Games Are Fun.txt"));
+gtest_expect_true(file_exists("Development Community.txt"));
+
+file_delete("Development Community.txt");
+gtest_expect_true(file_exists("Games Are Fun.txt"));
+gtest_expect_false(file_exists("Development Community.txt"));
+
+file_delete("Games Are Fun.txt");
+gtest_expect_false(file_exists("Games Are Fun.txt"));
+gtest_expect_false(file_exists("Development Community.txt"));
+gtest_expect_false(file_exists("ENIGMA John Doe.txt"));
+
+gtest_expect_eq(filename_name("C:/John/Doe/Smoe.txt"),"Smoe.txt");
+gtest_expect_eq(filename_path("C:/John/Doe/Smoe.txt"),"C:/John/Doe/");

```

Comment:
prefer `C:/Spot/Run`

Reply:
No, that's what filename_dir is for, the GM8.1 manual even says so.

```
filename_path(fname) Returns the path part of the indicated file name, including the final backslash.
filename_dir(fname) Returns the directory part of the indicated file name, which normally is the same as the path except for the final backslash.
```